astcontext/blocks.go

```diff
@@ -0,0 +1,221 @@
+package astcontext
+
+import (
+	"errors"
+	"go/ast"
+	"sort"
+)
+
+// Block represents anything, that contain block: Func, If, For, Range, Switch, TypeSwitch, Select.
+type Block struct {
+	BlockPos *Position `json:"block" vim:"block"`
+	Lbrace   *Position `json:"lbrace" vim:"lbrace"` // position of "{"
+	Rbrace   *Position `json:"rbrace" vim:"rbrace"` // position of "}"
+
+	node ast.Node
+}
+
+// Blocks represents a list of Block's.
+type Blocks []*Block
+
+// Blocks returns a list of Block's from the parsed source.
+func (p *Parser) Blocks() Blocks {
+	var files []*ast.File
+	if p.file != nil {
+		files = append(files, p.file)
+	}
+
+	if p.pkgs != nil {
+		for _, pkg := range p.pkgs {
+			for _, f := range pkg.Files {
+				files = append(files, f)
+			}
+		}
+	}
+
+	var blocks []*Block
+	inspect := func(n ast.Node) bool {
+		switch x := n.(type) {
+		case *ast.ForStmt:
```

Comment:
Cause in this case, we wouldn't be able to select block's declaration via `vab`, I mean select `if .....`, `for ....`.